src/value/jk_constant.rs

```diff
@@ -152,6 +159,12 @@ macro_rules! jk_primitive {
                 Some(self.to_instance())
             }
         }
+
+        impl TypeCheck for JkConstant<$t> {
+            fn resolve_type(&self, _: &mut Context) -> CheckedType {
+                CheckedType::Resolved(TypeId::from($s))
```

Comment:
Because this is done in a macro. You can look at the entire file, it explains the reason why this is necessary. `$s` is the string associated with the type, so we can also make the `TypeId` from it